mqclient/queue.py

```diff
@@ -219,6 +230,64 @@ def open_sub(self) -> "QueueSubResource":
         LOGGER.debug("Creating new QueueSubResource instance.")
         return QueueSubResource(self)
 
+    @contextlib.asynccontextmanager  # needs to wrap @wtt stuff to span children correctly
+    @wtt.spanned(
+        these=[
+            "self._broker_client",
+            "self._address",
+            "self._name",
+            "self._prefetch",
+            "self.timeout",
+        ]
+    )
+    async def open_sub_manual_acking(
+        self,
+    ) -> AsyncGenerator["ManualQueueSubResource", None]:
+        """Open a resource to receive messages from the queue as an iterator.
+
+        This returns a context-manager with an iterator function `iter_messages()`.
+        The iterator stops when no messages are received for `timeout` seconds.
+        Multiple calls to `open_sub()` is okay, but reusing the returned
+        instance is not.
+
+        The connection is closed after the context manager exits.
+
+        *Unlike `open_sub()`*, the caller is responsible for:
+            - All acking and/or nacking
+            - Any error handling
+
+        **NOTE: unless you need to parallelize your message processing,
+        use `open_sub()`**
+
+        Example:
+            async with queue.open_sub_manual_acking() as sub:
+                async for msg in sub.iter_messages():
+                    print(msg.data)
+                    sub.ack(msg)
+
+            async with queue.open_sub_manual_acking() as sub:
+                async for msg in sub.iter_messages():
+                    try:
+                        process_message(msg.data)
+                    except Exception:
+                        await gen.nack(msg)
+                    else:
+                        await gen.ack(msg)
```

Comment:
Where does `gen` come from in this example?

Reply:
good catch, copypasta